models/repo.go

```diff
@@ -2811,3 +2811,25 @@ func (repo *Repository) GetOriginalURLHostname() string {
 
 	return u.Host
 }
+
+// GetTreePathLock returns LSF lock for the treePath
+func (repo *Repository) GetTreePathLock(treePath string) (*LFSLock, error) {
```

Comment:
? unrelated?

Reply:
Coul be generated ba cherry-pick + conflict resolve ...

Reply:
done